File: runtime/jvm/src/lox/LoxBoundMethod.java
package lox;

/** A method closure paired with the receiver it was looked up on (GET_PROPERTY / GET_SUPER). */
public final class LoxBoundMethod implements LoxCallable {
    public final Object receiver;
    public final LoxClosure method;

    public LoxBoundMethod(Object receiver, LoxClosure method) {
        this.receiver = receiver;
        this.method = method;
    }

    @Override
    public Object call(Object[] args) {
        return method.callAsSelf(receiver, args);
    }
}

File: runtime/clr/src/LoxBoundMethod.cs
namespace Lox;

/// <summary>A method closure paired with the receiver it was looked up on (GET_PROPERTY / GET_SUPER).</summary>
public sealed class LoxBoundMethod : ILoxCallable {
    public readonly object Receiver;
    public readonly LoxClosure Method;

    public LoxBoundMethod(object receiver, LoxClosure method) {
        Receiver = receiver;
        Method = method;
    }

    public object Call(object[] args) => Method.CallAsSelf(Receiver, args);
}
